src/review.ts

```diff
@@ -0,0 +1,201 @@
+          ${this.truncationButtonsTemplate}
+        </article>`;
+  }
+
+  private get subjectTemplate(): string {
+    if (!this.review?.reviewtitle) return '';
+
+    return this.review.reviewtitle.length <= this.maxSubjectLength ||
+      this.showTruncatedContent
+      ? this.review.reviewtitle
+      : this.review.reviewtitle.slice(0, this.maxSubjectLength).concat('...');
+  }
+
+  private get bodyTemplate(): string {
+    if (!this.review?.reviewbody) return '';
+
+    return this.review.reviewbody.length <= this.maxBodyLength ||
+      this.showTruncatedContent
+      ? this.review.reviewbody
+      : this.review.reviewbody.slice(0, this.maxBodyLength).concat('...');
+  }
+
+  private get truncationButtonsTemplate(): HTMLTemplateResult | typeof nothing {
+    if (!this.review?.reviewtitle || !this.review?.reviewbody) return nothing;
+
+    const noTruncationNeeded =
+      this.review.reviewtitle.length <= this.maxSubjectLength &&
+      this.review.reviewbody.length <= this.maxBodyLength;
+
+    if (noTruncationNeeded) return nothing;
+
+    return this.showTruncatedContent
+      ? this.lessButtonTemplate
+      : this.moreButtonTemplate;
+  }
+
+  private get moreButtonTemplate(): HTMLTemplateResult {
+    return html`<button
+      class="simple-link more-btn"
+      @click=${() => (this.showTruncatedContent = true)}
+    >
+      ${msg('More...')}
+    </button>`;
+  }
+
+  private get lessButtonTemplate(): HTMLTemplateResult {
+    return html`<button
+      class="simple-link less-btn"
+      @click=${() => (this.showTruncatedContent = false)}
+    >
+      ${msg('...Less')}
+    </button>`;
+  }
+
+  private get reviewerTemplate(): HTMLTemplateResult | typeof nothing {
+    return !this.review
+      ? nothing
+      : this.review.itemname
+        ? html`<a
+            href="${this.baseHost}/details/${this.review.itemname}"
+            class="reviewer-link simple-link"
+            data-event-click-tracking="ItemReviews|ReviewerLink"
+            >${this.review.screenname}</a
+          >`
+        : html`${this.review.screenname}`;
+  }
+
+  private get starsTemplate(): HTMLTemplateResult | typeof nothing {
+    if (!this.review || !this.review.stars) return nothing;
+    return html`<div
+        class="review-stars"
+        title="${msg(`${this.review.stars} out of 5 stars`)}"
+      >
+        ${new Array(this.review.stars)
+          .fill(null)
+          .map(() => html`<div class="review-star">${starBasic}</div>`)}
+      </div>
+      - `;
+  }
+
+  private get createDateTemplate(): string | typeof nothing {
+    if (!this.review) return nothing;
+    const prettyDate = this.review.createdate?.toLocaleString('en-us', {
+      month: 'long',
+      day: 'numeric',
+      year: 'numeric',
+    });
+    const hasBeenEdited =
+      this.review.reviewdate?.getTime() !== this.review.createdate?.getTime();
+
+    return msg(`${prettyDate} ${hasBeenEdited ? '(edited)' : ''}`);
```

Comment:
for `${hasBeenEdited ? '(edited)' : ''}`, can we make a helper variable above it so the return value is simple